### PR TITLE
soc/arm/renesas_rzt2m: set default System Clock Control register values

### LINKED ISSUE #65674: boards: rzt2m_starterkit: Issue with serial output
**Describe the bug**
I am currently trying to get serial output from the "hello-world" application with the 'rzt2m_starterkit' board but I only seem to get junk from the USB serial connection (CN16). I have the board-rate set to "115200" which is correct according to this:

https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/arm/renesas/rzt2m.dtsi#L95

I have also used a logic analyser to try and debug this issue attaching a wire to the resistor for the IC39 line driver (can be found on schematic) on the back of the board which carries the signal "UART_USB_TX". From the signal I get the estimated board-rate of "152439". If the board-rate is set to this value you will still get some junk but it will show the "hello-world". This can be seen with the attached image below.

I'm unsure what's going wrong it could be that the some clocks are set incorrectly and are creating a non-standard board-rate in which the chip cannot read.  

image

**Expected behavior**
Working USB serial console  

**Environment**
 - OS: MacOS (have also tried on Linux)
 - Toolchain: Zephyr SDK 0.16.3

### PR DESCRIPTION
**Pull request by jbylicki**
This PR implements the verbose setting of default clock control register values for the RZ/T2M board. Previously, those registers were not set and if on the same `debugserver` session one has launched another binary and said registers were modified, it wouldn't re-adjust later in Zephyr.

Fixes #65674

### CODE REVIEW

**Review by tgorochowik — DISMISSED** [review]

**Comment by fkokosinski** [comment]
@joeker64 - could you take a look if this fix of the issue in #65674 works for you? Thanks

**Comment by fkokosinski** [comment]
Hi @aaillet! Could you take a look at this PR?

**Comment by joeker64** [comment]
> @joeker64 - could you take a look if this fix of the issue in #65674 works for you? Thanks

I can confirm this works now on my board

**Comment by fkokosinski** [comment]
> > > @joeker64 - could you take a look if this fix of the issue in #65674 works for you? Thanks
> > 
> > 
> > I can confirm this works now on my board
> 
> I tried it again today and now see no serial output. I've copied my steps from yesterday. Has anyone else experienced this?

Hi @joeker64! Are you sure you're using the default configuration of the on-board switches? Could you also elaborate on what was the difference in your setup between the first successful test and the second one? For the record, the on-board switches come with the default configuration as described by you in https://github.com/zephyrproject-rtos/zephyr/issues/65674#issuecomment-1845110625.

We've tested this on a brand new devkit which had no prior software uploaded before and the changes proposed in this PR indeed do fix the clock/baud rate issues, so we're not sure how we can reproduce the issue you're experiencing.

**Comment by fkokosinski** [comment]
Rebased & force-pushed due to conflicts.

**Review by tgorochowik — DISMISSED** [review]

**Comment by fkokosinski** [comment]
Hi @aaillet, could you take a look at this PR as an assignee? Thanks

**Comment by aaillet** [comment]
Hi,
Sorry but I'm not working with Renesas RZ boards, I don't have any documentation or boards to test any changes.
@tgorochowik that already approved these changes is Renesas RZ family maintainer since #67487 and should be the assignee for this PR.
Anyway, @fkokosinski tested this on board which is a relevant test, do we have any news from @joeker64 as he's the one that posted #65674 ?

**Comment by joeker64** [comment]
I haven't used the board since the last time. I can give it ago again tomorrow but I think it's more to do with my setup as it worked the first time I flashed it. Unsure what's changed from the first time I flashed it with this patch

**Comment by fkokosinski** [comment]
Rebased & force-pushed due to conflicts after merging the `collab-hwm` branch.

**Review by tgorochowik — APPROVED** [review]

**Review by kgugala — APPROVED** [review]
LGTM

**Pull request merged**
